lib/changes.js

```diff
@@ -103,23 +103,19 @@ export function removeNodes(selectorObserver, changes, nodes) {
 //
 // Returns nothing.
 export function revalidateObservers(selectorObserver, changes, node) {
-  if ('matches' in node) {
```

Comment:
So one thing this guards against is `nodeType`s like `TEXT_NODE`s that don't implement `matches`.